File: app/src/main/java/edu/und/seau/firebase/authentication/FirebaseAuthenticationInterface.java
package edu.und.seau.firebase.authentication;

import java.util.function.Predicate;

import androidx.core.util.Consumer;

public interface FirebaseAuthenticationInterface {

    void login(String email, String password, Consumer<Boolean> onResult);

    void register(String email, String password, String userName, Consumer<Boolean> onResult);

    String getUserId();
    String getUserName();
    void logOut(Consumer<Boolean> onResult);
}

File: app/src/main/java/edu/und/seau/firebase/authentication/FirebaseAuthenticationInterface.kt
package edu.und.seau.firebase.authentication

interface FirebaseAuthenticationInterface {

    fun login(email: String, password: String, onResult: (Boolean) -> Unit)
    fun register(username: String, password: String, email: String, onResult: (Boolean) -> Unit)
    fun getUserID(): String
    fun getUserName(): String
    fun logOut(onResult: () -> Unit)
}